### DISCUSSION TITLE
Team C - Describe components

### DISCUSSION THREAD

**Discussion by fregataa** [lablup/contribution-academy-2021#63] [Show and tell]
Describe the components of Backend.ai in one sentence.
[Link](https://github.com/lablup/contribution-academy-2021/wiki/Challengers:3%EC%A3%BC%EC%B0%A8#3-%EB%82%98%EB%A7%8C%EC%9D%98-%EB%AC%B8%EC%9E%A5%EC%9C%BC%EB%A1%9C-%EA%B0%81-%EC%BB%B4%ED%8F%AC%EB%84%8C%ED%8A%B8-%EC%84%A4%EB%AA%85%ED%95%B4%EB%B3%B4%EA%B8%B0)

**Comment by fregataa** [comment]
- common - backend.ai에서 전반적으로 사용할 수 있도록 만든 공용 파이썬 모듈
- client-py - 파이썬으로 구현된 backend.ai CLI
- kernels - backend.ai의 커널에서 사용될 이미지를 위한 저장소
- storage-proxy - backend.ai agent가 사용할 수 있는 가상 폴더 관리자

**Comment by euni00** [comment]
- Webserver : 웹 클라이언트에게 콘텐츠를 제공하는 서버이며 사용자 요청이 발생하면 상황에 따라 각각의 WAS에 요청을 전달하고, 그 처리를 다시 사용자에게 답변을 보내주는 역할을 수행한다.

- webUI :  기본 관리 및 사용자 모드 제공한다.

**Comment by yujung7768903** [comment]
Manager : 클러스터, 세션, 에이전트를 관리하며, 사용자의 요청에 따라 적절한 에이전트 및 세션으로 라우팅한다.
Agent : GPU가상화 및 보안 레이어를 제공하며, 컨테이너들을 관리하고 제어한다.

**Comment by jinseo-ok** [comment]
- backend.ai-client-py : CLI 환경에서의 입력을 backend.ai-manager로 통신하도록 관리 및 제어하는 python기반 컴포넌트
- backend.ai-manager : client-py로 부터 받은 request 를  처리하는 핵심 컴포넌트로, 클러스터의 연산자원 관리 및 세션 스케줄링, 사용자 인증 및 세션 실행 등의 API를 제공하는 컴포넌트
